File: visitor/ManageLeaveRequestsVisitor.java
package visitor;

public class ManageLeaveRequestsVisitor implements ScheduleManagementVisitor {
    @Override
    public void visit(DayShiftScheduleManagement dayShiftScheduleManagement) {
        System.out.println("Managing leave requests for day shift.");
    }

    @Override
    public void visit(NightShiftScheduleManagement nightShiftScheduleManagement) {
        System.out.println("Managing leave requests for night shift.");
    }

    @Override
    public void visit(RemoteWorkShiftScheduleManagement remoteWorkShiftScheduleManagement) {
        System.out.println("Managing leave requests for remote work shift.");
    }
}

File: C#/Visitor/ManageLeaveRequestsVisitor.cs
namespace Visitor;

public sealed class ManageLeaveRequestsVisitor : IScheduleManagementVisitor
{
    public void Visit(DayShiftScheduleManagement dayShiftScheduleManagement)
        => Console.WriteLine("Managing leave requests for day shift.");

    public void Visit(NightShiftScheduleManagement nightShiftScheduleManagement)
        => Console.WriteLine("Managing leave requests for night shift.");

    public void Visit(RemoteWorkShiftScheduleManagement remoteWorkShiftScheduleManagement)
        => Console.WriteLine("Managing leave requests for remote work shift.");
}
